### ISSUE TITLE
CrystalField cost function does not agree with Mantid const function

### ISSUE THREAD

**Issue by jfowkes** [fitbenchmarking/fitbenchmarking#914]
The CrystalField test problem accuracy reported by fitbenchmarking for the default _weighted_nlls_ cost does not agree with Mantid. According to Mantid the optimal fit has an accuracy of **1.27** c.f. the notebook here:
https://github.com/ralna/CrystalFieldOpt/blob/main/CrystalFieldExample.ipynb
whereas fitbenchmarking reports **269.7** (lm-scipy from scipy_ls).

**Comment by jfowkes** [comment]
Given fit data `(x,y,e)` and parameters `p` fitbenchmarking's default *weighted_nlls cost* is defined as
```
\sum_{i=1}^n ( [ y_i - f(x_i,p) ] / e_i )^2 
```
Digging into the Mantid source it seems that Mantid uses *Chi-squared over degrees of freedom* as the cost, defined as
```
chi**2 / [ len(y) - len(p) ]
```
for non-fixed parameters `p`. However this gives me **1.35** rather than the **1.27** that Mantid gives for CrystalField.

@Anders-Markvardsen do you know the exact formula for Mantid's *Chi-squared over degrees of freedom* and how exactly Mantid defines Chi-squared in terms of its default *Least-Squares cost*? Also is our *weighted_nlls cost* the same as Mantid's *Least-Squares cost*?

**Comment by AndrewLister-STFC** [comment]
Description is here: https://docs.mantidproject.org/nightly/algorithms/CalculateChiSquared-v1.html#algm-calculatechisquared

**Comment by jfowkes** [comment]
Reading through the various Mantid definitions of Chi-squared (both weighted/unweighted and with/without division by the degrees of freedom) it is clear that the way we have defined chi-squared in Fitbenchmarking is inconsistent. In the codebase we still refer to the cost function as chi-squared but this is simply not correct: chi-squared is a measure of the goodness of fit and therefore independent of the cost function used. I therefore propose we add a new Chi-squared results table that reports the four measures of Chi-squared that are used within Mantid as detailed in the above link.

**Comment by AndrewLister-STFC** [comment]
> 
> 
> Reading through the various Mantid definitions of Chi-squared (both weighted/unweighted and with/without division by the degrees of freedom) it is clear that the way we have defined chi-squared in Fitbenchmarking is inconsistent. In the codebase we still refer to the cost function as chi-squared but this is simply not correct: chi-squared is a measure of the goodness of fit and therefore independent of the cost function used. I therefore propose we add a new Chi-squared results table that reports the four measures of Chi-squared that are used within Mantid as detailed in the above link.

Yes, "Chi-Squared" was originally used correctly until we added more cost functions and the name didn't get updated. I'm not sure I agree that chi-squared is independant of the cost function, if we choose to minimise based on some very irregular weighting then it is possible that the best fit will be worse than some other fit using chi-squared. In fact in this case you would probably choose mantids "ChiSquaredWeighted" which matches the cost function.

I think changing the name and adding some kind of scaling option (e.g. the DOF one) are good ways forward, but I'm more hesitant about the chi-squared table.

**Comment by jfowkes** [comment]
> Yes, "Chi-Squared" was originally used correctly until we added more cost functions and the name didn't get updated. I'm not sure I agree that chi-squared is independant of the cost function, if we choose to minimise based on some very irregular weighting then it is possible that the best fit will be worse than some other fit using chi-squared. In fact in this case you would probably choose mantids "ChiSquaredWeighted" which matches the cost function.
> 
> I think changing the name and adding some kind of scaling option (e.g. the DOF one) are good ways forward, but I'm more hesitant about the chi-squared table.

So what I mean by *chi-squared is independent of the cost function* is that chi-squared is a particular goodness of fit measure which, while it coincides with certain cost functions, is not a cost function in and of itself. The issue is that we're currently mixing two different concepts here: a cost function that you optimise and a measure you use to evaluate the quality of a fit. I think that we should have the ability to report both of these in Fitbenchmarking (as is standard in the fitting community).

**Comment by AndrewLister-STFC** [comment]
From meeting: This would be good as entries in the support page for each fit

**Comment by jess-farmer** [comment]
@jess-farmer to triage
